src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -250,14 +250,14 @@ protected override void ComputeDependencyNodeDependencies(List<DependencyNodeCor
                 }
 
                 // We might have already compiled this method.
-                if (methodCodeNodeNeedingCode.StaticDependenciesAreComputed)
-                    continue;
+                if (methodCodeNodeNeedingCode.AlreadyStartedCompilation())
+                    return;
 
                 MethodDesc method = methodCodeNodeNeedingCode.Method;
                 if (!NodeFactory.CompilationModuleGroup.ContainsMethodBody(method, unboxingStub: false))
```

Comment:
Unrelated to this review, but since it's touching this code: @trylek Is this reachable? I wouldn't expect us to have a `MethodWithGCInfo` in the system for something that is not part of the version bubble.

Reply:
Please note that this in practice checks for the input module, not the version bubble, but I think you're right, the code in MethodEntrypoint explicitly performs this check to decide whether to emit a MethodWithGCInfo + LocalImport or an ExternalImport.